pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -150,6 +140,23 @@ func (c *Cache) InvalidateCachedPredicateItemForPodAdd(pod *v1.Pod, nodeName str
 		}
 	}
 	c.InvalidatePredicatesOnNode(nodeName, invalidPredicates)
+
+	// MatchInterPodAffinity: we assume scheduler can make sure newly bound pod
+	// will not break the existing inter pod affinity. So normally we does not need to
+	// invalidate MatchInterPodAffinity when pod added.
+	//
+	// NOTE: assumptions above will not stand when we implemented features like
+	// RequiredDuringSchedulingRequiredDuringExecutioc.
+	//
+	// But the exception is: pod has anti-affinity to itself. This is common in a ReplicaSet,
+	// so we have to invalidate MAtachInterPodAffinity in this case.
+	//
+	// TODO(harry): Should consider a more fine-grained way in the future, especially
+	// if eCache can be merged to predicates pkg.
+	invalidAntiAffinityPred := sets.NewString(predicates.MatchInterPodAffinityPred)
+	if pod.Spec.Affinity != nil && pod.Spec.Affinity.PodAntiAffinity != nil {
```

Comment:
Is there a similar issue for PodAffinity too?  The predicate on all other nodes except this one would have to be invalidated?

Reply:
Ecache data structure is like: ecache[node][pred][equivhash], for all the other nodes, they do not have cached info for this pod.